File: unitconverter.java
import java.util.Scanner;

public class UnitConverter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Unit Converter");
        System.out.println("1. Kilometers to Miles");
        System.out.println("2. Miles to Kilometers");
        System.out.print("Choose an option: ");
        int choice = scanner.nextInt();

        switch (choice) {
            case 1:
                System.out.print("Enter kilometers: ");
                double kilometers = scanner.nextDouble();
                double miles = kilometers * 0.621371;
                System.out.println(kilometers + " kilometers is equal to " + miles + " miles.");
                break;
            case 2:
                System.out.print("Enter miles: ");
                double milesInput = scanner.nextDouble();
                double kilometersResult = milesInput / 0.621371;
                System.out.println(milesInput + " miles is equal to " + kilometersResult + " kilometers.");
                break;
            default:
                System.out.println("Invalid choice.");
                break;
        }

        scanner.close();
    }
}

File: unitconverter.py
def display_menu():
    print("\nUnit Converter")
    print("1. Length (Meters to Kilometers)")
    print("2. Weight (Kilograms to Pounds)")
    print("3. Temperature (Celsius to Fahrenheit)")
    print("4. Exit")

def convert_length():
    meters = float(input("Enter length in meters: "))
    kilometers = meters / 1000
    print(f"{meters} meters is {kilometers} kilometers.")

def convert_weight():
    kilograms = float(input("Enter weight in kilograms: "))
    pounds = kilograms * 2.20462
    print(f"{kilograms} kilograms is {pounds} pounds.")

def convert_temperature():
    celsius = float(input("Enter temperature in Celsius: "))
    fahrenheit = (celsius * 9/5) + 32
    print(f"{celsius} Celsius is {fahrenheit} Fahrenheit.")

def main():
    while True:
        display_menu()
        choice = input("\nEnter your choice (1-4): ")

        if choice == '1':
            convert_length()
        elif choice == '2':
            convert_weight()
        elif choice == '3':
            convert_temperature()
        elif choice == '4':
            print("\nExiting Unit Converter. Goodbye!")
            break
        else:
            print("\nInvalid choice. Please select a valid option.")

if __name__ == "__main__":
    main()
